view/next-project/src/components/purchasereports/PurchaseReportItemNumModal.tsx

```diff
@@ -14,22 +14,38 @@ import {
 } from '@components/common';
 import PurchaseReportAddModal from '@components/purchasereports/PurchaseReportAddModal';
 import { useUI } from '@components/ui/context';
-import { PurchaseItem, PurchaseOrder, Expense } from '@type/common';
+import { PurchaseItem, PurchaseOrder, Expense, YearPeriod } from '@type/common';
 import { get } from '@utils/api/api_methods';
 
 export default function PurchaseReportItemNumModal() {
+  const date = new Date();
+  const [selectedYear, setSelectedYear] = useState<number>(date.getFullYear());
+  const [yearPeriods, setYearPeriods] = useState<YearPeriod[]>([]);
+  useEffect(() => {
+    const getPurchaseReportsUrl = process.env.CSR_API_URI + '/years/periods';
+    const getPeriods = async () => {
+      const res = await get(getPurchaseReportsUrl);
+      const year = res ? res[res.length - 1].year : date.getFullYear();
+      setSelectedYear(year);
+      setYearPeriods(res);
+    };
+    getPeriods();
+  }, []);
+
   const [user] = useRecoilState(userAtom);
   const [expenses, setExpenses] = useState<Expense[]>([]);
-  const [expenseID, setExpenseID] = useState(1);
+  const [expenseID, setExpenseID] = useState(0);
 
   useEffect(() => {
-    const getExpensesUrl = process.env.CSR_API_URI + '/expenses';
+    const getExpenseByPeriodsUrl =
+      process.env.CSR_API_URI + '/expenses/fiscalyear/' + String(selectedYear);
     const getExpenses = async () => {
-      const res = await get(getExpensesUrl);
+      const res = await get(getExpenseByPeriodsUrl);
       setExpenses(res);
+      setExpenseID(res[0].id);
```

Comment:
```suggestion
      if (res) {
        setExpenseID(res[0].id);
      }
```
![image](https://github.com/NUTFes/FinanSu/assets/106811268/973dd5b3-de9b-4f7c-b5a8-931e43a78fa2)
上記画像でnullの時にエラーが吐いていたのでそれの対処に応急処置みたいなコードです

Reply:
修正しました
[[fix]nullの時の処理追加](https://github.com/NUTFes/FinanSu/pull/716/commits/6f21a3ac75f997053bce367927ff1998f6ae88c0)
[Q[fix]購入報告で同じ物品が2個表示されているバグの修正](https://github.com/NUTFes/FinanSu/pull/716/commits/ca6cb89fa66cc150daf2dbb9fce83af2f89d34c4)